### ISSUE TITLE
parse_json() leaks memory

### ISSUE THREAD

**Issue by kortemik** [rsyslog/rsyslog#2391]
### Expected behavior
parse_json() should free memory after message is done
### Actual behavior
parse_json() does not free memory after message is done
### Steps to reproduce the behavior
configure rsyslog as follows:
```
main_queue(queue.type="Direct")
global(
    workDirectory="/opt/Fail-Safe/rsyslog/var"
    maxMessageSize="128k"
)
$MainMsgQueueType Direct
module(load="imrelp")
input(type="imrelp" port="601" MaxDataSize="128k")
set $.ret = parse_json("{ \"c1\":\"data\" }", "\$!parsed");
if ($hostname == "this.doesnt.exist") then {
    action(type="omfile" file="/test.log")
}
```
flood rsyslog with tcpflood until oom.
inspect during the flooding with strace -f -p $(pidof rsyslogd) 2>&1 | grep mprotect

### Environment
- rsyslog version: [e.g. 8.31.0 -- see rsyslogd -v]
```
rsyslogd 8.32.0, compiled with:
        PLATFORM:                               x86_64-pc-linux-gnu
        PLATFORM (lsb_release -d):
        FEATURE_REGEXP:                         Yes
        GSSAPI Kerberos 5 support:              No
        FEATURE_DEBUG (debug build, slow code): Yes
        32bit Atomic operations supported:      Yes
        64bit Atomic operations supported:      Yes
        memory allocator:                       system default
        Runtime Instrumentation (slow code):    No
        uuid support:                           Yes
        systemd support:                        No
        Number of Bits in RainerScript integers: 64

See http://www.rsyslog.com for more information.
```
- platform: [e.g. Ubuntu 16.04]
```
# cat /etc/redhat-release 
Fedora release 27 (Twenty Seven)
# uname -a
Linux 59ca53eb2397 3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```
- for configuration questions/issues, include rsyslog.conf:
included above
*Note: rsyslog documentation is available here*:
  [pre-release](http://www.rsyslog.com/doc/master/),
  [current stable release](http://www.rsyslog.com/doc/v8-stable/)

**Comment by kortemik** [comment]
We may have old version of libfastjson in use, and will verify this:
* libfastjson 0.99.8 is now required

**Comment by rgerhards** [comment]
thx for the great bug report and mentioning libfastjson lower than 0.99.8. I'll still try to see if that case is covered by the testbench and, if not, that I can add it. I'll report back when done.

**Comment by rgerhards** [comment]
confirmed via new testbench, fix underway

**Comment by rgerhards** [comment]
@kortemik https://github.com/rgerhards/rsyslog/commit/cdc727123249edb064057b32b8e1b1fdfd1cdd33 is the plain patch if you want to patch locally

**Comment by simnyc** [comment]
@kortemik has version 8.33 fixed this issue for you? 
I've upgraded but I'm still seeing increasing memory usage that leads to OOM.

**Comment by kortemik** [comment]
Yes it did, but we did backport it to 8.32 so i don't know about 8.33.

**Comment by rgerhards** [comment]
> @kortemik has version 8.33 fixed this issue for you?
> I've upgraded but I'm still seeing increasing memory usage that leads to OOM.

@simnyc do you use 8.33.0 or 8.33.1? The 8.33.0 tarball was bad, it did not contain the full 8.33.0 release.  A fixed version was re-released as 8.33.1.

**Comment by simnyc** [comment]
I was using `8.33.0-0adiscon1xenial1` from the PPA:

```
$ dpkg -l | grep rsyslog
ii  rsyslog                          8.33.0-0adiscon1xenial1                    amd64        a rocket-fast system for log processing
ii  rsyslog-mmjsonparse              8.33.0-0adiscon1xenial1                    amd64        Parsing/handling of CEE/Lumberjack JSON messages in rsyslog
ii  rsyslog-relp                     8.33.0-0adiscon1xenial1                    amd64        RELP protocol support for rsyslog
```

Thanks for pointing to 8.33.1, I've upgraded and will check the behavior.
